toolchain/sem_ir/value_stores.h

```diff
@@ -168,9 +168,22 @@ struct NameScope {
   // Names in the scope.
   llvm::DenseMap<NameId, InstId> names;
 
-  // Whether the scope corresponds to an import that failed. There may still be
-  // names from successful imports, or the current file.
-  bool has_load_error = false;
+  // Scopes extended by this scope.
+  //
+  // TODO: A `NameScopeId` is not sufficient to describe an extended scope in
+  // general. For example:
```

Comment:
One possibility here is that NameScopeId is extended to address generic arguments, either by providing a name scope for them or a variant of name scope that includes the parameter and points at the base name scope.

(i.e.,`NameScopeId` might still be the right choice since we do need to support generic name scopes in general)

Reply:
Yeah. I think we might want two layers here: a (nominal?) scope layer that has a name lookup table and a list of extended scopes, plus a (semantic?) scope layer that holds something like a reference to a nominal scope plus a list of generic arguments. But however we end up modeling it, our current representation (a `NameScopeId`, with its current meaning) isn't sufficient :-)

Reply:
I mean, if you change anything, I'd just say "currently". But I get the intent, and maybe just happened to read it in an overly prescriptive way.

```suggestion
  // TODO: A `NameScopeId` is currently insufficient to describe an extended scope in
  // general. For example:
```